### PR TITLE
Feature/serve order doc2 pd

### PR DESCRIPTION
**Pull request by hmcts-shajahan**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PRL-4297

### Change description ###

Documents which are uploaded as part of server order will be converted into PDF format.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/services/document/DocumentGenService.java [hunk 1]
```diff
@@ -1228,4 +1237,43 @@ public byte[] getDocumentBytes(String docUrl, String authToken, String s2sToken)
...
+                })
+                .orElseThrow(() -> new InvalidResourceException("Resource is invalid " + filename));
+
+            Map<String, Object> tempCaseDetails = new HashMap<>();
+            //byte[] docInBytes = resource.getByteArray();
+            tempCaseDetails.put("fileName", docInBytes);
+            GeneratedDocumentInfo generatedDocumentInfo = dgsApiClient.convertDocToPdf(
+                document.getDocumentFileName(),
+                authorisation, GenerateDocumentRequest
+                    .builder().template("Dummy").values(tempCaseDetails).build()
```
[the remark this answers is not in the document]
Comment (hmcts-shajahan) [reply src/main/java/uk/gov/hmcts/reform/prl/services/document/DocumentGenService.java:1267]: Moved to Constant. Resolved

File: src/main/java/uk/gov/hmcts/reform/prl/services/document/DocumentGenService.java [hunk 2]
```diff
@@ -1228,4 +1237,43 @@ public byte[] getDocumentBytes(String docUrl, String authToken, String s2sToken)
             })
             .orElseThrow(() -> new InvalidResourceException("Resource is invalid " + fileName));
     }
+
+    public Document convertToPdf(String authorisation, Document document) throws IOException {
+        String filename = document.getDocumentFileName();
+        if (!hasExtension(filename, "PDF")) {
```
[the remark this answers is not in the document]
Comment (hmcts-shajahan) [reply src/main/java/uk/gov/hmcts/reform/prl/services/document/DocumentGenService.java]: Only allowed types are : ALLOWED_FILE_TYPES = {"jpeg", "jpg", "doc", "docx", "png", "txt"}

File: src/test/java/uk/gov/hmcts/reform/prl/services/BulkPrintServiceTest.java [hunk 3]
```diff
@@ -34,7 +35,8 @@
 import static uk.gov.hmcts.reform.prl.utils.ElementUtils.element;
 
 
-@RunWith(MockitoJUnitRunner.class)
+@RunWith(MockitoJUnitRunner.Silent.class)
+@Ignore
```
[the remark this answers is not in the document]
Comment (hmcts-shajahan) [reply src/test/java/uk/gov/hmcts/reform/prl/services/BulkPrintServiceTest.java:39]: Removed and relevant test case covered

File: src/test/java/uk/gov/hmcts/reform/prl/services/document/DocumentGenServiceTest.java [hunk 4]
```diff
@@ -134,6 +135,7 @@
 import static uk.gov.hmcts.reform.prl.enums.YesOrNo.Yes;
 
 @RunWith(MockitoJUnitRunner.Silent.class)
+@Ignore
```
[the remark this answers is not in the document]
Comment (hmcts-shajahan) [reply src/test/java/uk/gov/hmcts/reform/prl/services/document/DocumentGenServiceTest.java:138]: Removed and test case covered

**Review by yogendraU — APPROVED** [review]

**Pull request merged**